### PR TITLE
Update `blog-starter` to App Router

### PR DESCRIPTION
**Pull request by zeevo**
### What?

This updates the `blog-starter` example to Next 14.1 App Router.

### Why?

I checked out a new `blog-starter` project on Vercel and was surprised it was using the Pages Router. I believe the App Router is a better choice.

### How?

I tried to keep the implementation logic as close to the original Pages `blog-starter`.

### CODE REVIEW

**Review by samcx — APPROVED** [review]
Thank you for submitting a <picture data-single-emoji=":pr:" title=":pr:">:pr:</picture>!

<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 19cec62fa55eebf3f11e24f8805e19622cf311d4

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/blog-starter/src/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+};
+
+export default function RootLayout({
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>) {
+  return (
+    <html lang="en">
+      <head>
```
Comment (zeevo) [inline examples/blog-starter/src/app/layout.tsx:25]: I believe this may add be adding a duplicate `<main>` tag. I can fix this later today.

Comment (samcx) [reply examples/blog-starter/src/app/layout.tsx:25]: @zeevo Thanks for pointing this out!

Comment (zeevo) [reply examples/blog-starter/src/app/layout.tsx:25]: @samcx Fixed it now with latest (squashed) commit 👍

**Review by samcx — APPROVED** [review]
<picture data-single-emoji=":lgtm:" title=":lgtm:">:lgtm:</picture>

**Comment by samcx** [comment]
@zeevo Only maintainers are allowed to approve CI runs so I went ahead and approved your commit again. There was an issue with the `package.json` that was causing a lint issue, so made some changes there!

**Comment by zeevo** [comment]
@samcx Woot! Thank you for the merge

**Pull request merged**